kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/benchmarks/WorkloadSpec.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.defaults.benchmarks;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import java.nio.file.Path;
+
+/** Base class for all workload specifications. */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
+@JsonSubTypes({@JsonSubTypes.Type(value = ReadMetadata.class, name = "read_metadata")})
+public abstract class WorkloadSpec {
```

Comment:
any reason you went for abstract class rather than interface?

Reply:
> edit: ah, probably cuz of the jackson stuff with the type

Correct :) 

Should I promote the comments below to be class docs?
```
// The type of workload (e.g., "read_metadata") This is used by Jackson's polymorphic
  // deserialization to automatically instantiate the correct subclass based on the "type" field in
  // JSON.
```